system-tests/__snapshots__/devServer_dynamic_support.ts.js

```diff
@@ -0,0 +1,18 @@
+exports['devServer dynamic import throws error when devServer use dynamic import instead of require 1'] = `
+The \`component\`.\`devServer\` method must be a function with the following signature:
+
+\`\`\`
+devServer: (cypressConfig: DevServerConfig, devServerConfig: ComponentDevServerOpts) {
```

Comment:
This is inconsistent with the arguments that are documented around `devServer`, which are:

```js
devServer(cypressDevServerConfig, devServerConfig) {}
```

Separately, I'm not sure the `ComponentDevServerOpts` type is entirely accurate, our dev server plugins define their own types for `devServerConfig`, and none of them seem to extend `ComponentDevServerOpts`. For example: 

```
❯ rg 'function devServer' -g '*.d.ts'
npm/react/plugins/react-scripts/index.d.ts
21:  function devServer(cypressDevServerConfig: Cypress.DevServerConfig, devServerConfig?: CypressCRADevServerConfig): Cypress.ResolvedDevServerConfig

npm/react/plugins/next/index.d.ts
15:  function devServer(cypressDevServerConfig: Cypress.DevServerConfig, devServerConfig?: CypressNextDevServerConfig): Cypress.ResolvedDevServerConfig

npm/react/plugins/babel/index.d.ts
25:  function devServer(cypressDevServerConfig: Cypress.DevServerConfig, devServerConfig?: CypressBabelDevServerConfig): Cypress.ResolvedDevServerConfig

npm/react/plugins/load-webpack/index.d.ts
21:  function devServer(cypressDevServerConfig: Cypress.DevServerConfig, devServerConfig?: CypressWebpackDevServerConfig): Cypress.ResolvedDevServerConfig

npm/react/plugins/craco/index.d.ts
21:  function devServer(cypressDevServerConfig: Cypress.DevServerConfig, devServerConfig: CypressCracoDevServerConfig): Cypress.ResolvedDevServerConfig
```